### PR TITLE
release-20.2: sql: support UPDATE ONLY syntax

### LINKED ISSUE #53552: sql: support UPDATE ONLY syntax
In Postgres, you can write a query like `UPDATE ONLY tablename ...`. Here's what the docs say:

>  If ONLY is specified before the table name, matching rows are updated in the named table only. If ONLY is not specified, matching rows are also updated in any tables inheriting from the named table. Optionally, * can be specified after the table name to explicitly indicate that descendant tables are included.

https://www.postgresql.org/docs/10/sql-update.html

I've only encountered this in one test case from the PGJDBC test suite.

Perhaps this would be a candidate for "good first issue" -- I'm adding to the SQL Features triage board so someone with more knowledge can decide to add that label.

### PR DESCRIPTION
**Pull request by angelapwen**
Backport 1/1 commits from #54277.

/cc @cockroachdb/release

---

Resolves #53552. 

Release note (sql change): Added the ability to parse the ONLY and * (descendant) clauses in UPDATE. This is a no-op as we do not support table inheritance.

### CODE REVIEW

**Comment by otan** [comment]
This can be merged when release team approves it. Then you can click the green merge button.

**Review by solongordon — APPROVED** [review]

**Pull request merged**
